code/modules/surgery/blood_filter.dm

```diff
@@ -0,0 +1,61 @@
+/datum/surgery/blood_filter
+	name = "Filter blood"
+	steps = list(/datum/surgery_step/incise,
+				/datum/surgery_step/retract_skin,
+				/datum/surgery_step/incise,
+				/datum/surgery_step/filter_blood,
+				/datum/surgery_step/close)
+
+	target_mobtypes = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
+	possible_locs = list(BODY_ZONE_CHEST)
+	requires_bodypart_type = TRUE
+	ignore_clothes = FALSE
+	var/antispam = FALSE
+
+/datum/surgery_step/filter_blood
+	name = "Filter blood"
+	implements = list(/obj/item/blood_filter = 95)
+	repeatable = TRUE
+	time = 2.5 SECONDS
+
+/datum/surgery/blood_filter/can_start(mob/user, mob/living/carbon/target)
+	if(HAS_TRAIT(target, TRAIT_HUSK) || target.reagents.total_volume==0) //Can't filter husk or 0 regent body
+		return FALSE
+	return ..()
+
+/datum/surgery_step/filter_blood/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(istype(surgery,/datum/surgery/blood_filter))
+		var/datum/surgery/blood_filter/the_surgery = surgery
+		if(!the_surgery.antispam)
+			display_results(user, target, "<span class='notice'>You begin filtering [target]'s blood...</span>",
+		"<span class='notice'>[user] uses [tool] to filtering your blood.</span>",
```

Comment:
this line makes me want to die

Reply:
i don't know enough about coding to know why

Reply:
“Uses blood filter to filtering your blood”

Reply:
ahf uck i see it now